### DISCUSSION TITLE
Searching for beatmaps with a large library is laggy

### DISCUSSION THREAD

**Discussion by Toasterly** [ppy/osu#25669] [Q&A] [answered]
I know this was asked 2 years ago with the answer "We will optimize this properly", but was wondering if anything will be changing on this anytime soon before ranked or should I just delete 70,000 of my maps? 
Would like to switch to Lazer for the upcoming release of ranked but the indexing freezes so bad its impossible to type or select a star range to find a map without wanting to tear my hair out.

https://github.com/ppy/osu/assets/37195727/f14d15f9-c37e-4227-a2de-1d28f2dbdc04

**Comment by marioman63** [comment]
Not sure if it will help in the meantime, but as someone with 200,000 maps in Lazer, I find moving them to a SSD helps a lot. Definitely improved my query times beyond what you demonstrate.

**Reply by peppy** [reply]
That sounds weird, the query is in-memory so should not change with an SSD.

**Comment by peppy** [comment]
This looks like it may have regressed, it was never anywhere near this bad.

**Reply by peppy** [reply, marked as answer]
Improved via https://github.com/ppy/osu/pull/25679
